components/datadog/agentparams/params.go

```diff
@@ -117,7 +120,19 @@ func WithAgentConfig(config string) func(*Params) error {
 // WithIntegration adds the configuration for an integration.
 func WithIntegration(folderName string, content string) func(*Params) error {
 	return func(p *Params) error {
-		p.Integrations[folderName] = content
+		confPath := path.Join("conf.d", folderName, "conf.yaml")
+		p.Files[confPath] = content
+		return nil
+	}
+}
+
+// WithFile adds a file to the install with the contents at the given path
```

Comment:
What do you think about having `WithFilemanager`? `WithFile` could use `WithFilemanager` for its implementation.
`FileManager` supports absolute paths, relative paths, embedded paths and `sudo` and I am afraid our user will try to convert everything to absolute paths to use `WithFile` instead of using existing function. What do you think?
Note: Can be done in a separate PR.